frontend/src/components/Utils/filter-input.test.js

```diff
@@ -24,14 +24,18 @@ describe('getCleanedTextInput tests', () => {
 });
 
 const mockCurUserEmail = 'cur.user@email.com';
+jest.mock('../AuthUtils', () => ({
+    getCurUserEmail: () => mockCurUserEmail,
+}));
 // TODO(Issue #55): Replace mock with real auth file once integrated.
 jest.mock('./temp-auth-utils.js', () => ({
-    getCurUserEmail: () => mockCurUserEmail,
-    getUserUidFromUserEmail: (userEmail) => '_' + userEmail + '_',
+    getUserUidArrFromUserEmailArr: (userEmailArr) => {
+          return userEmailArr.map(userEmail => '_' + userEmail + '_');
+        },
 }));
 describe('getCollaboratorUidArray tests', () => {
   test('No collaborators entered', () => {
-    const expectedUidArr = [getUserUidFromUserEmail(mockCurUserEmail)];
+    const expectedUidArr = getUserUidArrFromUserEmailArr([mockCurUserEmail]);
```

Comment:
It would be better to hardcode the expected array here, since isn't the code that is called (getCollaboratorUidArray) ultimately just calling your mock function? Separating the expected value from the code run and simplifying tests so they're easy to understand and debug is preferred.

Reply:
`getCollaboratorUidArray` does have some other logic in it but for this test case, yes they both call that mock function. 

So I have a question when it comes to testing then (this applies for the other related comments). When you create the expected value, is it recommended that in pretty much every case that the expected value should be hardcoded? And if that is the case, is it okay to include variable declarations of the test inputs in the expected value? For example, in the code you reference below ("Same added collaborators"), it is okay to use the variables `person1Email` and `person2Email` in expected value like how I changed it?

Otherwise I totally get what you are saying and that makes sense. Check out my fixes and let me know if anything else seems off.

Reply:
The way you did it is fine. The most important thing is to decouple the expected values from the functions that will be run to produce them (since you want to verify that those functions _do indeed_ get called and produce them.

Reply:
Ahh okay. I think the lines got blurred for me when the functions used in the function being tested were mocked and the expected value was simply the mock function with the test values passed into it. But that makes more sense now, thank you!